qualtran/drawing/flame_graph.py

```diff
@@ -0,0 +1,132 @@
+        return _pretty_name(val)
+    if isinstance(val, float):
+        if np.isclose(val, 0):
+            val = 0
+        return f'{val:0.2g}'
+    return f'{val}'
+
+
+def _pretty_name(bloq: Bloq) -> str:
+    from qualtran.serialization.bloq import _iter_fields
+
+    ret = bloq.pretty_name()
+    if bloq.pretty_name.__qualname__.startswith('Bloq.'):
+        for field in _iter_fields(bloq):
+            ret += f'[{_pretty_arg(getattr(bloq, field.name))}]'
+    return ret
+
+
+@functools.lru_cache(maxsize=None)
+def _t_counts_for_bloq(bloq: Bloq, graph: nx.DiGraph) -> int:
+    sigma = _compute_sigma(bloq, graph)
+    return t_counts_from_sigma(sigma)
+
+
+def _keep_if_small(bloq: Bloq) -> bool:
+    from qualtran.bloqs.basic_gates import Toffoli, TwoBitCSwap
+    from qualtran.bloqs.mcmt.and_bloq import And
+
+    if isinstance(bloq, (And, Toffoli, TwoBitCSwap)):
+        return True
+
+
+def _populate_flame_graph_data(
+    bloq: Bloq, graph: nx.DiGraph, graph_t: nx.DiGraph, prefix: List[str]
+) -> List[str]:
+    from qualtran.bloqs.basic_gates import TGate
+
+    callees = [x for x in list(graph.successors(bloq)) if _t_counts_for_bloq(x, graph_t) > 0]
+    total_t_counts = _t_counts_for_bloq(bloq, graph_t)
+    prefix.append(_pretty_name(bloq) + f'(T:{total_t_counts})')
+    data = []
+    if len(callees) == 0 or (len(callees) == 1 and callees[0] in [TGate(), TGate(is_adjoint=True)]):
+        data += [';'.join(prefix) + '\t' + str(total_t_counts)]
+    else:
+        succ_t_counts = 0
+        for succ in callees:
+            curr_data = _populate_flame_graph_data(succ, graph, graph_t, prefix)
+            succ_t_counts += (
+                _t_counts_for_bloq(succ, graph_t) * graph.get_edge_data(bloq, succ)['n']
+            )
+            data += curr_data * graph.get_edge_data(bloq, succ)['n']
+        # assert total_t_counts == succ_t_counts, f'{bloq=}, {total_t_counts=}, {succ_t_counts=}'
+    prefix.pop()
+    return data
+
+
+def get_flame_graph_data(
+    *bloqs: Bloq,
+    file_path: Union[None, pathlib.Path, str] = None,
+    keep: Optional[Sequence['Bloq']] = _keep_if_small,
+    **kwargs,
+) -> List[str]:
+    from qualtran.resource_counting.generalizers import cirq_to_bloqs
+
+    data = []
+    for bloq in bloqs:
+        call_graph, _ = bloq.call_graph(keep=keep, **kwargs, generalizer=cirq_to_bloqs)
+        call_graph_t_counts, _ = bloq.call_graph()
+        data += _populate_flame_graph_data(bloq, call_graph, call_graph_t_counts, prefix=[])
+    if file_path:
+        with open(file_path, 'w') as f:
+            f.write('\n'.join(data))
+    else:
+        return data
+
+
+def get_flame_graph_svg_data(
+    *bloqs: Bloq, file_path: Union[None, pathlib.Path, str] = None, **kwargs
+) -> Optional[str]:
+    data = get_flame_graph_data(*bloqs, **kwargs)
+
+    data_file = tempfile.NamedTemporaryFile(mode='w')
+    flame_graph_path = pathlib.Path(__file__).resolve().parent.parent / "third_party/flamegraph.pl"
```

Comment:
I'm pretty sure this won't work if you `pip install qualtran`.

Reply:
Do you have suggestions for what to change? can we fix this in a followup PR?